amgut/lib/data_access/ag_data_access.py

```diff
@@ -303,28 +303,39 @@ def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
                      JOIN ag_consent USING (ag_login_id, participant_name)
                      WHERE ag_login_id = %s AND participant_name = %s"""
             TRN.add(sql, (ag_login_id, participant_name))
-            survey_id, participant_email = TRN.execute_fetchindex()[0]
+            # collect all survey_ids and participant_names, since at least the
+            # former might be more than one.
+            survey_ids = set()
+            participant_emails = set()
+            for hit in TRN.execute_fetchindex():
+                survey_ids.add(hit[0])
+                participant_emails.add(hit[1])
 
             sql = "DELETE FROM survey_answers WHERE survey_id = %s"
-            TRN.add(sql, [survey_id])
+            for survey_id in survey_ids:
```

Comment:
Even it is a single transaction the number of sql queries changes and each sql queries has some overhead. Reducing the number of SQL queries reduces the running time of the method.